### DISCUSSION TITLE
阿里翻译注册使用教程

### DISCUSSION THREAD

**Discussion by YiiGuxing** [YiiGuxing/TranslationPlugin#1208] [Q&A] [answered]
# 阿里翻译注册使用教程

> 阿里翻译的注册使用在版本的发行说明里已经有说明的，但是由于插件网站的Pages托管站点 `github.io` 和 `gitee.io` 目前都无法正常访问，所以在这里也放一个简单的教程吧。

1. 注册阿里云账号并开通[通用版翻译引擎](https://www.aliyun.com/product/ai/base_alimt)；
2. 在[访问凭证管理](https://ram.console.aliyun.com/manage/ak)页面获取AccessKey ID（应用程序ID）和AccessKey Secret（应用程序密钥）；
3. 在插件的阿里翻译配置页面注册应用程序ID和应用程序密钥即可。

**Comment by homeant** [comment]
是不是需要花钱

**Comment by YiiGuxing** [comment, marked as answer]
> 是不是需要花钱？

有免费配额的，提供每月一百万字符的免费配额，满足日常需求：
https://yiiguxing.gitee.io/translation-plugin/updates.html?v=3.2

**Comment by prczhb** [comment]
是不是必须用阿里的主账号的 AccessKey ID 和 AccessKey Secret, 使用子账号的AccessKey ID 和 AccessKey Secret 一直提示没权限.
![image](https://user-images.githubusercontent.com/26354648/142127968-1e74f48a-150e-458a-9e93-6a90404497cc.png)
关于子账号, 我已经给了这些权限了啊, 仍然提示没权限

还是照着官方的文档来的 权限和他一模一样
https://help.aliyun.com/document_detail/96396.html?spm=a2c4g.11186623.6.574.87b1372ff7wGju

**Comment by prczhb** [comment]
> 是不是必须用阿里的主账号的AccessKey ID和AccessKey Secret，使用子账号的AccessKey ID和AccessKey Secret一直提示没权限。 关于子账号，我已经给了这些权限了啊，还是提示没权限 ![图片](https://user-images.githubusercontent.com/26354648/142127968-1e74f48a-150e-458a-9e93-6a90404497cc.png)
> 
> 还是照着官方的文档来的权限和他一模一样 https://help.aliyun.com/document_detail/96396.html?spm=a2c4g.11186623.6.574.87b1372ff7wGju

有毒, 用阿里云app又设置了一遍权限, 发现和电脑端的权限不一样...有一个全资源权限
![image](https://user-images.githubusercontent.com/26354648/142128854-41217d5a-78cd-4233-8efc-b482e988bb8c.png)

**Comment by PhoenixZeng** [comment]
~~右键 翻译选中的单词 -> 正常
右键 翻译文档 -> "子账号没有权限"
 (已经授予了全资源的 翻译管理 和 只读访问 两个权限)
怎么解决?~~

好像是授权有延迟引起的.现在好了 

不过还有一个问题:
使用阿里翻译时 会出现 "检测语言" 状态 导致无法播放语音
![image](https://user-images.githubusercontent.com/26225903/159170192-a68bd0bc-1f4d-4062-8d96-16110398e75c.png)
只有手动改为英语后 才可以播放
对比google
![image](https://user-images.githubusercontent.com/26225903/159170165-d5f92ee5-5fa9-4748-a137-7c7df5318fb6.png)

**Comment by YiiGuxing** [comment]
> 使用阿里翻译时 会出现 "检测语言" 状态 导致无法播放语音
> 只有手动改为英语后 才可以播放

@PhoenixZeng 这是因为在使用 "检测语言" 时，阿里翻译并没有返回检测出的语言，没有语言是无法进行语音播放的。

**Comment by wangyufei123** [comment]
我这边使用百度，阿里，有道的翻译时连不上网络

**Comment by kang1097** [comment]
<img width="221" alt="截屏2023-01-11 14 52 15" src="https://user-images.githubusercontent.com/110171491/211741494-e61931db-726b-4ab6-8779-3e0440740938.png">
你好，最新版本的Translation，idea是2021.1的，配置的阿里翻译，一直显示这个，我用微软翻译是好的，我记得阿里翻译有免费配额的，是阿里账户里必须要有金额才能使用免费配额吗？

**Reply by YiiGuxing** [reply]
应该是你的免费配额使用完了，你到阿里的控制台看看？

**Reply by kang1097** [reply]
<img width="1237" alt="image" src="https://user-images.githubusercontent.com/110171491/211748243-3efef09b-74d7-4546-8a55-32f6fd4fca43.png">
谢谢，好了，需要把这两个开通一下，就行了。
